ui/components/app/wallet-overview/eth-overview.js

```diff
@@ -118,6 +120,36 @@ const EthOverview = ({ className, showAddress }) => {
   }
 
   const isSwapsChain = useSelector(getIsSwapsChain);
+  const signingEnabled = useSelector((state) =>
+    checkIfMethodIsEnabled(state, EthMethod.SignTransaction),
+  );
+
+  const buttonTooltips = {
+    sendButton: [{ condition: !signingEnabled, message: 'accountCannotSign' }],
+    swapButton: [
+      { condition: !isSwapsChain, message: 'currentlyUnavailable' },
+      { condition: !signingEnabled, message: 'accountCannotSign' },
+    ],
+    bridgeButton: [
+      ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
+      { condition: !isBridgeChain, message: 'currentlyUnavailable' },
+      ///: END:ONLY_INCLUDE_IF
+      { condition: !signingEnabled, message: 'accountCannotSign' },
+    ],
+  };
+
+  const generateTooltip = (buttonKey, contents) => {
+    const conditions = buttonTooltips[buttonKey];
+    const tooltipInfo = conditions.find(({ condition }) => condition);
+    if (tooltipInfo) {
+      return (
+        <Tooltip title={t(tooltipInfo.message)} position="bottom">
+          {contents}
+        </Tooltip>
+      );
+    }
+    return contents;
+  };
```

Comment:
Good improvement! My only question now is code coverage, are we testing this properly? Either at an unit or E2E level

Reply:
According to the code coverage report, we are at 85% of the lines covered in the `eth-overview.js` with all the new lines of code and functions added in this PR being 100% covered. The function to generate the tooltip and the tool tips themselves are tested at a minimum of 31x each in the current unit testing suite (see images below). Is there a specific scenario you think an e2e test would encompass that we are missing here? 
<img width="1907" alt="Screenshot 2024-02-29 at 11 17 08 AM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/46eda819-2964-4b11-a7eb-e33de1b1b823">
<img width="651" alt="Screenshot 2024-02-29 at 11 22 33 AM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/0488854b-a404-4fbb-a982-ed6ce5bb0b31">
<img width="532" alt="Screenshot 2024-02-29 at 11 23 32 AM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/1daf473d-36aa-49d0-8a09-ed5a41b8a73f">
<img width="514" alt="Screenshot 2024-02-29 at 11 24 07 AM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/5c44db11-f9ed-4b21-b3fb-c842cacc80bd">
<img width="506" alt="Screenshot 2024-02-29 at 11 24 17 AM" src="https://github.com/MetaMask/metamask-extension/assets/91970214/5ad91f60-e766-4544-9648-3633c827a3a8">

Reply:
That looks good enough, thanks for checking!